### DISCUSSION TITLE
ComfyUI Support

### DISCUSSION THREAD

**Discussion by jkovacs-dev** [AbdBarho/stable-diffusion-webui-docker#367] [Ideas]
Is there any interest/support for adding a [ComfyUI ](https://github.com/comfyanonymous/ComfyUI) profile?  I played around with it for about an hour and was able to get a proof of concept up and running fairly easily.  (although would definitely need to invest some time to properly figure out all the mount point mappings since they're structured a bit differently than any of the current profiles)

**Comment by krokosik** [comment]
Did you do anything special compared to their Readme? I'm also trying to add this profile and it seems that everything installs correctly, but when I run the GUI on port 7860 it says in the logs that it started, but the address is dead :(

**Comment by jkovacs-dev** [comment]
Not really, no. It did take quite some time to start though. Several minutes in fact. I believe it checks all the models on startup so if you have more than a few, it can take a while to read them all. You can check out my branch [here](https://github.com/jkovacs-dev/stable-diffusion-webui-docker/tree/comfyui).

**Comment by AbdBarho** [comment]
I have nothing against the idea, I am gonna have to see how it is going to work with the license https://github.com/comfyanonymous/ComfyUI/blob/master/LICENSE

maybe I will need to update the license of this repo as well.

I am intending on doing some clean up of this repo first, I will comeback to this idea soon.

**Reply by jkovacs-dev** [reply]
That’s a good point! I hadn’t even considered the license! Let me know what you decide.

**Reply by jkovacs-dev** [reply]
Also, is there anything you want help with in cleaning up the repo? Happy to help.

**Reply by AbdBarho** [reply]
Thank you! I am busy with stuff from another repo, [Open Assistant](https://github.com/LAION-AI/Open-Assistant). We could always use a helping hand there, if thats something for you.

**Comment by PassiveLemon** [comment]
I'm going to write this down for future reference:
I can't test if this is the Docker implementation or ComfyUI itself, but after a prompt finishes, i get an error having to do with `FileExistsError: [Errno 17] File exists: '/stable-diffusion/output`
I can manually fix this by going into the container and deleting the output "file" and making an output directory but this isn't very practical.
The full error is here https://pastebin.com/AaRHQQYC

I tried adding a `mkdir -p /stable-diffusion/output/` line in the Dockerfile for comfy but that did not seem to do anything.

EDIT 2:
Solved. [PR here](https://github.com/jkovacs-dev/stable-diffusion-webui-docker/pull/1)
